### ISSUE TITLE
🐛 cancelDeviceConnection exception, kill the app

### ISSUE THREAD

**Issue by simonitfi** [dotintent/react-native-ble-plx#1171]
### Prerequisites

- [X] I checked the documentation and FAQ without finding a solution
- [X] I checked to make sure that this issue has not already been filed

### Expected Behavior

In general all errors should throw Exception on JavaScript side so we could handle exceptions on react native.

We have similar issues on other methods as well, throwing random exceptions that could not be handled.

### Current Behavior

Exceptions that could not be handled. And kills the app.

### Library version

3.1.2

### Device

Android 12

### Environment info

```shell
System:
  OS: Windows 11 10.0.22000
  CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  Memory: 5.69 GB / 15.89 GB
Binaries:
  Node:
    version: 20.10.0
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: ~\AppData\Roaming\npm\yarn.CMD
  npm:
    version: 10.2.3
    path: C:\Program Files\nodejs\npm.CMD
  Watchman:
    version: 20230430.125247.0
    path: C:\Android\watchman.EXE
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    Versions:
      - 10.0.19041.0
IDEs:
  Android Studio: AI-223.8836.35.2231.10406996
  Visual Studio:
    - 17.0.31903.59 (Visual Studio Community 2022)
Languages:
  Java:
    version: 17.0.3.1
    path: C:\Ohjelmointi\jdk-17.0.3.1\bin\javac.EXE
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.6
    wanted: 0.72.6
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Steps to reproduce

bleManager.cancelDeviceConnection(id)

### Formatted code sample or link to a repository

```js
-
```

### Relevant log output

```shell
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
       at bleshadow.dagger.internal.Preconditions.checkNotNull(Preconditions.java)
       at com.polidea.rxandroidble2.DaggerClientComponent$DeviceComponentBuilder.macAddress(DaggerClientComponent.java)
       at com.polidea.rxandroidble2.DaggerClientComponent$DeviceComponentBuilder.macAddress(DaggerClientComponent.java)
       at com.polidea.rxandroidble2.internal.RxBleDeviceProvider.getBleDevice(RxBleDeviceProvider.java:42)
       at com.polidea.rxandroidble2.RxBleClientImpl.getBleDevice(RxBleClientImpl.java:5)
       at com.bleplx.adapter.BleModule.cancelDeviceConnection(BleModule.java:18)
       at com.bleplx.BlePlxModule.cancelDeviceConnection(BlePlxModule.java:17)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:148)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(:147)
       at com.facebook.jni.NativeRunnable.run(SourceFile)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage()
       at android.os.Looper.loopOnce(Looper.java:233)
       at android.os.Looper.loop(Looper.java:344)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(:37)
       at java.lang.Thread.run(Thread.java:1012)
```

**Comment by EmmaZachara** [comment]
Thank you for reporting this issue. We are analyzing it and will get back to you with more info soon.

**Comment by stefanpieter** [comment]
Hey Guys

We are having the same on iOS.

Please have a look at this :-)

**Comment by EmmaZachara** [comment]
Hi @stefanpieter, @simonitfi. We are still working on this and some other issues, so please hold on a bit longer.

**Comment by intent-kacper-cyranowski** [comment]
Hi everyone

The issue with the NullPointerException was related to calling a method on non-existent objects. This has been resolved in version 3.2.0.

**Comment by intent-kacper-cyranowski** [comment]
I’m closing the issue, in case of anyone with similar error please upgrade to at least 3.2.0 version
